### PR TITLE
add FieldGroup, take two

### PR DESCRIPTION
**Pull request by TomGoBravo**
Adds a new enum which groups CommonFields into categories. Currently these are tests, vaccines, hospitals, cases_deaths.

This PR reverts https://github.com/covid-projections/covid-data-public/pull/209

## Tested

* `./run.py data update` in covid-data-model repo passed
* Added some more unit tests in both repos which reproduced the problem introduced by 208

### CODE REVIEW

File: covidactnow/datapublic/common_fields.py [hunk 1]
```diff
@@ -28,103 +28,127 @@ class FieldName(str):
...
+    def __new__(cls, field_name: str, field_group: Optional[FieldGroup]):
+        """Make an enum that is both a str and has a field_group attribute."""
+        # TODO(tom): When we are using Python >= 3.8 incorporate backports.strenum, per
+        #  https://discuss.python.org/t/built-in-strenum/4192
+        # Could be replaced with use of https://pypi.org/project/typeguard/
+        assert isinstance(field_name, str)
+        assert field_group is None or isinstance(field_group, FieldGroup)
+        # This solution is hacked from https://stackoverflow.com/a/49286691
+        o = super().__new__(cls, field_name)
+        o._value_ = field_name
```
Comment (ghop02) [inline covidactnow/datapublic/common_fields.py:52]: Is there something special about `_value_`? I browsed the SO and didn't quite grasp the issue

**Review by ghop02 — APPROVED** [review]
did tests pass on covid-data-model with this code?

**Comment by TomGoBravo** [comment]
> did tests pass on covid-data-model with this code?

Yes. The passed with #208 too :-/ so https://github.com/covid-projections/covid-data-model/pull/960 adds tests that would have failed with #208 and pass with this change.

File: covidactnow/datapublic/common_fields.py (hunk 1 above)
Comment (TomGoBravo) [reply covidactnow/datapublic/common_fields.py:54]: Added another comment with d809bdd

**Pull request merged**